xdsl/dialects/affine.py

```diff
@@ -25,12 +25,31 @@
     irdl_op_definition,
     operand_def,
     opt_attr_def,
+    prop_def,
     region_def,
     result_def,
     var_operand_def,
     var_result_def,
 )
 from xdsl.traits import IsTerminator
+from xdsl.utils.exceptions import VerifyException
+
+
+@irdl_op_definition
+class Apply(IRDLOperation):
+    name = "affine.apply"
+
+    mapOperands = var_operand_def(IndexType)
+    map = prop_def(AffineMapAttr)
+    result = result_def(IndexType)
+
+    def verify_(self) -> None:
+        if len(self.mapOperands) != self.map.data.num_dims + self.map.data.num_symbols:
+            raise VerifyException(
+                "affine.apply expects as many operands as its map's dimensions and symbols."
```

Comment:
Can we make this error message more descriptive? Something like (adjust accordingly):

```suggestion
                f"{self.name} expects {self.map.data.num_dims + self.map.data.num_symbols} operands, but got len(self.mapOperands). The number of map operands must match the sum of the dimensions and symbols of its map."
```

Reply:
Same for below